File: Core/Entities/Invoice.cs
using System;
using System.Collections.Generic;
using Core.Entities;


namespace Core.Entities
{
    public class Invoice : BaseEntity
    {
        public Invoice() 
        {
            
        }

        public Invoice(Customer customer, int customerId, DateTimeOffset date, string workPerformed, decimal subtotal, decimal tPS5, decimal tVQ9975, ICollection<InvoiceItem> items, string appliance, string modelNumber, string submitter)
        {
            Customer = customer;
            CustomerId = customerId;
            Date = date;
            WorkPerformed = workPerformed;
            Subtotal = subtotal;
            TPS5 = tPS5;
            TVQ9975 = tVQ9975;
            Items = items;
            Appliance = appliance;
            ModelNumber = modelNumber;
            Submitter = submitter;
        }

        public Customer Customer { get; set; }
        public int CustomerId { get; set; }
        public DateTimeOffset Date { get; set; } = DateTimeOffset.Now;
        public string WorkPerformed { get; set; }
        public decimal Subtotal { get; set; }
        public decimal TPS5 { get; set; } 
        public decimal TVQ9975 { get; set; }
        public ICollection<InvoiceItem> Items { get; set; }
        public string Appliance { get; set; }
        public string ModelNumber { get; set; }
     
        public string Submitter { get; set; }

        public decimal GetTotal()
        {
            return Subtotal + TPS5 + TVQ9975;
        }
    }
}

File: client/src/app/shared/models/invoice.ts
import { ICustomer } from "./customer";


export interface IInvoice {
    id: number;
    customer: ICustomer;
    customerId: number;
    Date: string;
    workPerformed: string;
    appliance: string;
    modelNumber: string;
    subTotal: number;
    tps5: number;
    tvq9975: number;
    items: IInvoiceItem[];
    total: number;
    submitter: string;
}

export interface IInvoiceItem {
    id: number;
    invoice: IInvoice;
    invoiceId: number;
    name: string;
    price: number;
    quantity: number;
}

export class InvoiceItem {
    id: number;
    invoice: IInvoice;
    invoiceId: number;
    name: string;
    price: number;
    quantity: number;
}

export class Invoice implements IInvoice {
    id: number;
    customer : ICustomer;
    customerId: number;
    Date : string;
    workPerformed : string;
    subTotal: number;
    tps5: number;
    tvq9975: number;
    items: IInvoiceItem[] = [];
    total: number;
    appliance: string;
    modelNumber: string;
    submitter: string;
}